src/deterministic.jl

```diff
@@ -1,76 +1,83 @@
 """
-Construct Deterministic from a SortedDict of Arrays.
-
-# Arguments
-- `name::AbstractString`: user-defined name
-- `input_data::AbstractDict{Dates.DateTime, Vector{Float64}}`: time series data.
-- `resolution::Dates.Period`: The resolution of the forecast in Dates.Period`
-- `normalization_factor::NormalizationFactor = 1.0`: optional normalization factor to apply
-  to each data entry
-- `scaling_factor_multiplier::Union{Nothing, Function} = nothing`: If the data are scaling
-  factors then this function will be called on the component and applied to the data when
-  [`get_time_series_array`](@ref) is called.
+A deterministic forecast for a particular data field in a Component.
 """
+mutable struct Deterministic <: Forecast
+    forecast::Union{DeterministicStandard, DeterministicSingleTimeSeries}
+end
+
+#=
+#  Design Note
+#  The Deterministic struct abstracts the two possible types of deterministic forecasts.
+#  1. DeterministicStandard stores vectors of forecasted values by incrementing initial time
+#     where each vector is of length `horizon`.
+#  2. DeterministicSingleTimeSeries stores a single vector of values by wrapping a
+#     SingleTimeSeries instance. It can be viewed as a forecast by taking slices of the
+#     vector at incrementing offsets.
+#
+#  Both types implement the same interfaces, and so they are interchangeable for the user.
+#  Selection of one type vs the other will happen during the parsing process based on the
+#  type of input data.
+#
+#  The Deterministic abstraction makes it so the normal user (modeller) will not know or
+#  care what the underlying type is.
+#
+#  A consequence of this abstraction is that the Deterministic struct must implement every
+#  constructor and method that the subtypes implement and forward to them. This causes
+#  quite a bit of duplication. A different solution would have prevented that. We could
+#  have created an AbstractDeterministic and made DeterministicStandard and
+#  DeterministicSingleTimeSeries subtypes of it, and then returned those specific types to
+#  the user. The team decided that this was overly complicated for the target users.
+=#
+
 function Deterministic(
     name::AbstractString,
-    input_data::AbstractDict{Dates.DateTime, Vector{Float64}};
     resolution::Dates.Period,
```

Comment:
Making resolution a required argument here needs to be made in Probabilistic too.